core/block.ts

```diff
@@ -1102,29 +1100,37 @@ export class Block implements IASTNodeLocation {
           ' instead',
       );
     }
-    for (let i = 0, input; (input = this.inputList[i]); i++) {
-      for (let j = 0, field; (field = input.fieldRow[j]); j++) {
-        if (field.name === name) {
-          return field;
-        }
+    for (const field of this.getFields()) {
+      if (field.name === name) {
+        return field;
       }
     }
     return null;
   }
 
+  /**
```

Comment:
Add a link to https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Generator to clarify that this is generator-in-javascript-language and not Blockly-Javascript-generator.

Reply:
Done